### PR TITLE
Modify delete command

### PR DESCRIPTION
**Pull request by Ragnapop**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -21,29 +25,70 @@ public class DeleteCommand extends Command {
...
+                throw new CommandException(Messages.MESSAGE_NON_EXISTENT_PERSON);
+            }
 
-        model.deletePerson(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
+            model.deletePerson(personToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
+        }
+        else {
+            List<Person> peopleToDelete = new ArrayList<Person>() {};
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:70]: Perhaps consider removing the curly brackets as they are not needed

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -58,13 +103,27 @@ public boolean equals(Object other) {
         }
 
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
-        return nusId.equals(otherDeleteCommand.nusId);
+        if (this.nusId != null) {
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:106]: Remove the ```this``` keyword as it is not necessary

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -58,13 +103,27 @@ public boolean equals(Object other) {
...
+            return nusId.equals(otherDeleteCommand.nusId);
+        }
+        else if (this.group != null) {
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:109]: Same here

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 4]
```diff
@@ -58,13 +103,27 @@ public boolean equals(Object other) {
...
+        }
+        return false;
     }
 
     @Override
     public String toString() {
-        return new ToStringBuilder(this)
-                .add("targetnusId", nusId.toString())
-                .toString();
+        if (this.nusId != null) {
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:117]: A few more ```this``` keyword found

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 5]
```diff
@@ -58,13 +103,27 @@ public boolean equals(Object other) {
...
+            return new ToStringBuilder(this)
+                    .add("targetnusId", nusId.toString())
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:119]: Naming convention should be ```targetNusId```

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 6]
```diff
@@ -58,13 +103,27 @@ public boolean equals(Object other) {
...
+                    .toString();
+        }
+        else if (this.group != null) {
+            return new ToStringBuilder(this)
+                    .add("targetgroup", group.toString())
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:124]: A few more naming convention errors

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java [hunk 7]
```diff
@@ -17,9 +20,30 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
...
+
+            }
+
+            else {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            }
+
+
+
+
```
Comment (Rikko1204) [inline src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:46]: Perhaps consider ensuring no duplicate prefix here?

**Review by Rikko1204 — COMMENTED** [review]
Overall, LGTM 👍

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 8]
```diff
@@ -14,10 +15,10 @@ box Model MODEL_COLOR_T1
 participant "m:Model" as Model MODEL_COLOR
 end box
 
-[-> LogicManager : execute("delete 1")
+[-> LogicManager : execute("delete E01234567")
```
Comment (cheahTJ) [inline docs/diagrams/DeleteSequenceDiagram.puml:18]: Remember to update this to `delete id/E01234567` to reflect the change in the `DeleteCommand` :)

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 9]
```diff
@@ -21,29 +25,69 @@ public class DeleteCommand extends Command {
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the NusId used in the displayed person list.\n"
             + "Parameters: NusId (8 digits long, starting with an 'E'). \n"
-            + "Example: " + COMMAND_WORD + " E0123456";
+            + "Example: " + COMMAND_WORD + " id/E0123456\n\n"
+            + "OR Deletes all the people identified by the group used in the displayed person list.\n"
+            + "Parameters: Group (Already existing in the address book). \n"
+            + "Example: " + COMMAND_WORD + " g/CS2103-T15";
```
Comment (cheahTJ) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:31]: Might want to consider listing down all the functionalities of the command, then provide the usage (examples) together afterwards.

**Pull request merged**